backend/de.metas.manufacturing/src/main/java/org/eevolution/productioncandidate/service/produce/PPOrderAllocatorService.java

```diff
@@ -0,0 +1,139 @@
+
+	@NonNull
+	private final PPOrderCandidatePojoConverter ppOrderCandidateConverter;
+
+	public PPOrderAllocatorService(final @NonNull PPOrderCandidatePojoConverter ppOrderCandidateConverter)
+	{
+		this.ppOrderCandidateConverter = ppOrderCandidateConverter;
+	}
+
+	@NonNull
+	public PPOrderAllocator buildAllocator(@NonNull final PPOrderCandidateToAllocate ppOrderCandidateToAllocate)
+	{
+		final Quantity capacity = getCapacityPerProductionCycle(ppOrderCandidateToAllocate.getPpOrderCandidate());
+
+		final String headerAggKey = ppOrderCandidateToAllocate.getHeaderAggregationKey();
+
+		final PPOrderCreateRequest.PPOrderCreateRequestBuilder requestBuilder = createRequestBuilder(ppOrderCandidateToAllocate.getPpOrderCandidate());
+
+		return PPOrderAllocator.builder()
+				.ppOrderCreateRequestBuilder(requestBuilder)
+				.capacityPerProductionCycle(capacity)
+				.headerAggKey(headerAggKey)
+				.build();
+	}
+
+	@NonNull
+	private PPOrderCreateRequest.PPOrderCreateRequestBuilder createRequestBuilder(@NonNull final I_PP_Order_Candidate candidateRecord)
+	{
+		final PPOrderCandidate ppOrderCandidatePojo = ppOrderCandidateConverter.toPPOrderCandidate(candidateRecord);
+
+		final ProductId productId = ProductId.ofRepoId(ppOrderCandidatePojo.getPpOrderData().getProductDescriptor().getProductId());
+
+		return PPOrderCreateRequest.builder()
+				.clientAndOrgId(ppOrderCandidatePojo.getPpOrderData().getClientAndOrgId())
+				.productPlanningId(ProductPlanningId.ofRepoId(ppOrderCandidatePojo.getPpOrderData().getProductPlanningId()))
+				.materialDispoGroupId(ppOrderCandidatePojo.getPpOrderData().getMaterialDispoGroupId())
+				.plantId(ppOrderCandidatePojo.getPpOrderData().getPlantId())
+				.warehouseId(ppOrderCandidatePojo.getPpOrderData().getWarehouseId())
+				//
+				.productId(productId)
+				.attributeSetInstanceId(AttributeSetInstanceId.ofRepoIdOrNone(ppOrderCandidatePojo.getPpOrderData().getProductDescriptor().getAttributeSetInstanceId()))
+				//
+				.dateOrdered(SystemTime.asInstant())
+				.datePromised(ppOrderCandidatePojo.getPpOrderData().getDatePromised())
+				.dateStartSchedule(ppOrderCandidatePojo.getPpOrderData().getDateStartSchedule())
+				//
+				.salesOrderLineId(OrderLineId.ofRepoIdOrNull(ppOrderCandidatePojo.getPpOrderData().getOrderLineIdAsRepoId()))
+				.shipmentScheduleId(ShipmentScheduleId.ofRepoIdOrNull(ppOrderCandidatePojo.getPpOrderData().getShipmentScheduleIdAsRepoId()))
+				//
+				.packingMaterialId(ppOrderCandidatePojo.getPpOrderData().getPackingMaterialId())
+				//dev-note: there is a custom logic for completing PPOrder when created from PP_Order_Candidate (due to MD_Candidates interaction)
+				//see: org.eevolution.productioncandidate.service.produce.PPOrderProducerFromCandidate.processPPOrderCandidates
+				.completeDocument(false);
+	}
+
+	@NonNull
+	private Quantity getCapacityPerProductionCycle(@NonNull final I_PP_Order_Candidate ppOrderCandidate)
```

Comment:
(not sure) i think at the end of this method is a good place to fallback to PP_Product_PLanning in case the resource has no CapacityPerProductionCycle defined.

Reply:
not needed (seee req. changes)